### ISSUE TITLE
Uncategorized error for from_unixtime

### ISSUE THREAD

**Issue by martint** [prestodb/presto#6550]
```
Invalid offset minutes 1080
java.lang.IllegalArgumentException
    com.facebook.presto.spi.type.TimeZoneKey.checkArgument(TimeZoneKey.java:300)
    com.facebook.presto.spi.type.TimeZoneKey.getTimeZoneKeyForOffset(TimeZoneKey.java:140)
    com.facebook.presto.spi.type.DateTimeEncoding.packDateTimeWithZone(DateTimeEncoding.java:41)
    com.facebook.presto.operator.scalar.DateTimeFunctions.fromUnixTime(DateTimeFunctions.java:149)
    com_facebook_presto_$gen_PageProcessor_14559.project_1(Unknown Source)
    com_facebook_presto_$gen_PageProcessor_14559.process(Unknown Source)
    com.facebook.presto.operator.FilterAndProjectOperator.getOutput(FilterAndProjectOperator.java:111)
    com.facebook.presto.operator.Driver.processInternal(Driver.java:378)
    com.facebook.presto.operator.Driver.processFor(Driver.java:301)
    com.facebook.presto.execution.SqlTaskExecution$DriverSplitRunner.processFor(SqlTaskExecution.java:622)
    com.facebook.presto.execution.TaskExecutor$PrioritizedSplitRunner.process(TaskExecutor.java:534)
    com.facebook.presto.execution.TaskExecutor$Runner.run(TaskExecutor.java:670)
    java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
    java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
    java.lang.Thread.run(Thread.java:745)
```

**Comment by martint** [comment]
@cberner, I don't think this is fixed, yet. Why did you close it?

**Comment by martint** [comment]
I just checked. It's still an issue:

```sql
select from_unixtime(0,0,841);
```

```
Query 20170105_010406_00002_buym6 failed: Invalid offset minutes 841
java.lang.IllegalArgumentException: Invalid offset minutes 841
	at com.facebook.presto.spi.type.TimeZoneKey.checkArgument(TimeZoneKey.java:300)
	at com.facebook.presto.spi.type.TimeZoneKey.getTimeZoneKeyForOffset(TimeZoneKey.java:140)
	at com.facebook.presto.spi.type.DateTimeEncoding.packDateTimeWithZone(DateTimeEncoding.java:41)
	at com.facebook.presto.operator.scalar.DateTimeFunctions.fromUnixTime(DateTimeFunctions.java:149)
	at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:627)
	at java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:649)
	at com.facebook.presto.sql.planner.ExpressionInterpreter.invoke(ExpressionInterpreter.java:1359)
	at com.facebook.presto.sql.planner.ExpressionInterpreter$Visitor.visitFunctionCall(ExpressionInterpreter.java:962)
	at com.facebook.presto.sql.tree.FunctionCall.accept(FunctionCall.java:111)
	at com.facebook.presto.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at com.facebook.presto.sql.planner.ExpressionInterpreter.optimize(ExpressionInterpreter.java:271)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions$Rewriter.simplifyExpression(SimplifyExpressions.java:155)
	at com.facebook.presto.sql.planner.plan.Assignments.lambda$rewrite$1(Assignments.java:111)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at com.facebook.presto.sql.planner.plan.Assignments.rewrite(Assignments.java:112)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions$Rewriter.visitProject(SimplifyExpressions.java:110)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions$Rewriter.visitProject(SimplifyExpressions.java:88)
	at com.facebook.presto.sql.planner.plan.ProjectNode.accept(ProjectNode.java:91)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:85)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:96)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions$Rewriter.visitProject(SimplifyExpressions.java:109)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions$Rewriter.visitProject(SimplifyExpressions.java:88)
	at com.facebook.presto.sql.planner.plan.ProjectNode.accept(ProjectNode.java:91)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:85)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:96)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions$Rewriter.visitProject(SimplifyExpressions.java:109)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions$Rewriter.visitProject(SimplifyExpressions.java:88)
	at com.facebook.presto.sql.planner.plan.ProjectNode.accept(ProjectNode.java:91)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:85)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:96)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions$Rewriter.visitProject(SimplifyExpressions.java:109)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions$Rewriter.visitProject(SimplifyExpressions.java:88)
	at com.facebook.presto.sql.planner.plan.ProjectNode.accept(ProjectNode.java:91)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.rewrite(SimplePlanRewriter.java:85)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.lambda$defaultRewrite$0(SimplePlanRewriter.java:74)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Iterator.forEachRemaining(Iterator.java:116)
	at java.util.Spliterators$IteratorSpliterator.forEachRemaining(Spliterators.java:1801)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter$RewriteContext.defaultRewrite(SimplePlanRewriter.java:75)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter.visitPlan(SimplePlanRewriter.java:39)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter.visitPlan(SimplePlanRewriter.java:23)
	at com.facebook.presto.sql.planner.plan.PlanVisitor.visitOutput(PlanVisitor.java:50)
	at com.facebook.presto.sql.planner.plan.OutputNode.accept(OutputNode.java:81)
	at com.facebook.presto.sql.planner.plan.SimplePlanRewriter.rewriteWith(SimplePlanRewriter.java:28)
	at com.facebook.presto.sql.planner.optimizations.SimplifyExpressions.optimize(SimplifyExpressions.java:85)
	at com.facebook.presto.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:120)
	at com.facebook.presto.sql.planner.LogicalPlanner.plan(LogicalPlanner.java:111)
	at com.facebook.presto.execution.SqlQueryExecution.doAnalyzeQuery(SqlQueryExecution.java:292)
	at com.facebook.presto.execution.SqlQueryExecution.analyzeQuery(SqlQueryExecution.java:271)
	at com.facebook.presto.execution.SqlQueryExecution.start(SqlQueryExecution.java:229)
	at com.facebook.presto.execution.QueuedExecution.lambda$start$1(QueuedExecution.java:62)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by cberner** [comment]
Oh sorry, I saw that https://github.com/prestodb/presto/pull/6706 was closed, but didn't scroll down to see that it was abandoned, not merged

**Comment by haozhun** [comment]
In order to fix this, `TimeZoneKey.checkArgument` needs to be changed to throw a subclass of `IllegalArgumentException`. All Presto functions that may hit this need to catch it and throw a PrestoException with proper error code (which could be different for different functions).

**Comment by synhershko** [comment]
@haozhun `TimeZoneKey.checkArgument` throws `IllegalArgumentException`, so I guess what's left to do is catch and rethrow as PrestoException from all consumers?

**Comment by nezihyigitbasi** [comment]
@yingsu00 will work on this.

**Comment by nezihyigitbasi** [comment]
Fixed with #9618.
